src/query.ts

```diff
@@ -24,5 +33,8 @@ export function useQuery<
     onUnmounted(() => view.value.destroy())
   }
 
-  return computed(() => view.value.data)
+  return {
+    data: computed(() => view.value.data),
+    resultDetails: computed(() => ({ type: view.value.resultType })),
```

Comment:
maybe

```suggestion
    resultType: computed(() => ({ type: view.value.resultType })),
```

Reply:
`type` is part of the result details and in the future there might be more details? I don't know what the rocicorp team is planning but that is how I interpreted it

Reply:
I've thought about it and maybe we should indeed keep the api as flat as possible like you suggested. I've asked around in the Rocicorp discord for their opinion (and their ideas on the api for react and solid)

Reply:
How about this?
```suggestion
    resultType: computed(() => view.value.resultType),
```
or even
```suggestion
    status: computed(() => view.value.resultType),
```
because `resultType` indicates the sync status of the query
- `complete` the query is fully in sync with the source db
- `unknown` showing cached data and checking / in progress of sync